src/Analyzers/CSharp/Analyzers/UseIndexOrRangeOperator/CSharpUseRangeOperatorDiagnosticAnalyzer.cs

```diff
@@ -111,13 +111,54 @@ protected override void InitializeWorker(AnalysisContext context)
                 }
             }
 
-            // look for `s.Slice(e1, end - e2)`
-            if (invocation.Instance is null ||
-                invocation.Arguments.Length != 2)
+            // look for `s.Slice(e1, end - e2)` or `s.Slice(e1)`
+            if (invocation.Instance is null)
             {
                 return null;
             }
 
+            return invocation.Arguments.Length switch
+            {
+                1 => AnalyzeInvocationWithOneArgument(invocation, infoCache, invocationSyntax, option),
+                2 => AnalyzeInvocationWithTwoArguments(invocation, infoCache, invocationSyntax, option),
+                _ => null,
+            };
+        }
+
+        private static Result? AnalyzeInvocationWithOneArgument(
+            IInvocationOperation invocation,
+            InfoCache infoCache,
+            InvocationExpressionSyntax invocationSyntax,
+            CodeStyleOption2<bool> option)
+        {
+            // See if the call is to something slice-like.
+            var targetMethod = invocation.TargetMethod;
+
+            // Try to see if we're calling into some sort of Slice method with a matching
+            // indexer or overload
```

Comment:
i'm not finding tehse comments helpful.  having real world examples of the before/after would be more helpful here imo.

Reply:
Agreed. I re-worded the comment.